### PR TITLE
Add events and update logs

### PR DESCRIPTION
**Pull request by VaishnaviHire**
This PR adds following changes:

- Add events to the controller
- Fix stream of `*unstructured. Unstructured resource` errors
- Update logs in the controller

Testing Steps:
- Deploy custom operator image as given in [Operator.md](https://github.com/opendatahub-io/opendatahub-operator/blob/master/operator.md)
- Deploy an example Kfdef 
-  Watch events in the `Events` tab in the cluster with filter for resource `KfDef`
-  Watch logs in `logs` tab for operator pod

### CODE REVIEW

File: pkg/controller/kfdef/const.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ const (
 
 var (
 	// watchedResources contains all resources we will watch and reconcile when changed
-	watchedResources = []schema.GroupVersionKind{
+	WatchedResources = []schema.GroupVersionKind{
 		{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "ClusterRole"},
```
Comment (nakfour) [inline pkg/controller/kfdef/const.go:15]: Is there a technical reason for changing from small to capital "W" ?

File: pkg/controller/kfdef/const.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ var (
 		{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "ValidatingWebhookConfiguration"},
 	}
 
-	watchedKubeflowResources = []schema.GroupVersionKind{
+	WatchedKubeflowResources = []schema.GroupVersionKind{
 		{Group: "app.k8s.io", Version: "v1beta1", Kind: "Application"},
```
Comment (nakfour) [inline pkg/controller/kfdef/const.go:40]: Same as previous comment, why capital "W"

File: pkg/controller/kfdef/kfdef_controller.go [hunk 3]
```diff
@@ -245,11 +250,11 @@ var ownedResourcePredicates = predicate.Funcs{
 		if err != nil {
 			return false
 		}
-		log.Infof("Got update event for %v.%v.", object.GetName(), object.GetNamespace())
 		// if this object has an owner, let the owner handle the appropriate recovery
```
Comment (nakfour) [inline pkg/controller/kfdef/kfdef_controller.go:253]: Why did you delete this log info? I know we see a lot of them. but sometimes it helps to debug

File: pkg/controller/kfdef/kfdef_controller.go [hunk 4]
```diff
@@ -402,6 +416,8 @@ func (r *ReconcileKfDef) Reconcile(request reconcile.Request) (reconcile.Result,
 	err = getReconcileStatus(instance, kfApply(instance))
 	if err == nil {
 		log.Infof("KubeFlow Deployment Completed.")
+		r.recorder.Eventf(instance, v1.EventTypeNormal, "KfDefCreationSuccessful",
```
Comment (nakfour) [inline pkg/controller/kfdef/kfdef_controller.go:419]: Same questions as above, do we need the log.info?

File: pkg/controller/kfdef/kfdef_controller.go [hunk 5]
```diff
@@ -616,10 +634,13 @@ func (r *ReconcileKfDef) operatorUninstall(request reconcile.Request) error {
...
 						return fmt.Errorf("error deleting namespace %v: %v", namespace.Name, err)
 					}
+					r.recorder.Eventf(&namespace, v1.EventTypeNormal, "NamespaceDeletionSuccessful",
+						"Namespace %s deleted as a part of uninstall.", namespace.Name )
 					log.Infof("Namespace %s deleted as a part of uninstall.", namespace.Name)
 				}
 			}
 		}
+	log.Info("All resources deleted as part of uninstall. Removing the operator csv")
 	return removeCsv(r.client, r.restConfig)
```
[the remark this answers is not in the document]
Comment (VaishnaviHire) [reply pkg/controller/kfdef/kfdef_controller.go:644]: `log.Info` is used when we only want to log a statement. `log.Infof` is used for formatted logs. 
For eg: 
```
log.Infof("Kfdef instance %v created, instance.Name)
```

File: cmd/manager/main.go [hunk 6]
```diff
@@ -169,3 +180,37 @@ func serveCRMetrics(cfg *rest.Config) error {
...
+func filterGKVsFromAddToScheme(gvks []schema.GroupVersionKind) []schema.GroupVersionKind {
+	matchAnyValue := "*"
+	watchedGVKs := append(kfdefcontroller.WatchedKubeflowResources, kfdefcontroller.WatchedResources...)
+
+	ownGVKs := []schema.GroupVersionKind{}
+	for _, gvk := range gvks {
+		for _, watchedGVK := range watchedGVKs {
+			match := true
+			if watchedGVK.Kind == matchAnyValue && watchedGVK.Group == matchAnyValue && watchedGVK.Version == matchAnyValue {
+				match = false
```
[the remark this answers is not in the document]
Comment (VaishnaviHire) [reply cmd/manager/main.go:198]: Yes it should be true. It ensures that we only add the `gvks` that do not match any of the conditions defined in `if/else`
